### PR TITLE
Document host endpoint support for eBPF mode

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

* Host endpoints now supported.
* Add OpenShift kube-proxy instructions.
* Sync up formatting with enterprise version of doc to reduce conflicts later.

## Related issues/PRs

documents https://github.com/projectcalico/felix/pull/2664

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: maintenance/ebpf/enabling-bpf.md
Comment (lwr20) [inline maintenance/ebpf/enabling-bpf.md:59]: FYI, https://wiki.ubuntu.com/BionicBeaver/ReleaseNotes#Kernel

> Ubuntu 18.04.4 ships with a v5.3 based Linux kernel updated from the v5.0 based kernel in 18.04.3.

Dunno if you want to update the 20.04 bullet with that?

**Review by lwr20 — APPROVED** [review]
One note, otherwise LGTM

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
